### PR TITLE
[DROOLS-4646] Add support for ? placeholder in unary expressions in DMN scenarios

### PR DESCRIPTION
**Pull request by danielezonca**
This PR introduce `?` as variable to refer actual value in unary expressions (EXPECT) so it will be possible to write expression like `max(1, ?) > 1` or similar.

NOTE: `?` is a special identifier in FEEL and the only one we can use in an unary test

@yesamer @gitgabrio @dupliaka 

https://issues.jboss.org/browse/DROOLS-4646

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java [hunk 1]
```diff
@@ -45,6 +45,8 @@ public void evaluateUnaryExpression() {
         assertTrue(expressionEvaluator.evaluateUnaryExpression("not( true )", false, boolean.class));
         assertTrue(expressionEvaluator.evaluateUnaryExpression(">2, >5", BigDecimal.valueOf(6), BigDecimal.class));
         assertTrue(expressionEvaluator.evaluateUnaryExpression("abs(-1)", BigDecimal.valueOf(1), BigDecimal.class));
+        assertTrue(expressionEvaluator.evaluateUnaryExpression("max(1, ?) > 1", BigDecimal.valueOf(2), BigDecimal.class));
+        assertTrue(expressionEvaluator.evaluateUnaryExpression("? = 2", BigDecimal.valueOf(2), BigDecimal.class));
```
Comment (gitgabrio) [inline drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: @danielezonca 
Can not evaluate the code - anyway I think it would be safer to add some "assertFalse" or whatever test - I mean not something that throws an exception, but something that simply does not returns the expected value.

**Review by gitgabrio — CHANGES_REQUESTED** [review]

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java (hunk 1 above)
Comment (danielezonca) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: Done :+1:

**Review by gitgabrio — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

File: drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/utils/ConstantsHolder.java [hunk 2]
```diff
@@ -24,6 +24,9 @@
     public static final String MVEL_ESCAPE_SYMBOL = "#";
     public static final String ACTUAL_VALUE_IDENTIFIER = "actualValue";
 
+    /* FEEL constants */
+    public static final String UNARY_EXPRESSION_IDENTIFIER = "?";
```
Comment (dupliaka) [inline drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/utils/ConstantsHolder.java]: Suggestion: we  can name it something like UNARY_PARAMETER_IDENTIFIER. UNARY_EXPRESSION_IDENTIFIER it looks that I should place ? before each unary expression

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java (hunk 1 above)
Comment (dupliaka) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: Also I m not sure, but probably it could be better to add a test where we passing a null in ?
like
assertThatThrownBy(() -> expressionEvaluator.evaluateUnaryExpression("? > 2", null, BigDecimal.class)))
                .isInstanceOf(IllegalArgumentException.class)

Comment (dupliaka) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: Also we can add a test where we can use as ? as we want (at last two) to declare that it is allowed

**Review by dupliaka — APPROVED** [review]
Approved with suggestions

**Comment by dupliaka** [comment]
jenkins execute full downstream build

**Comment by dupliaka** [comment]
jenkins execute full downstream build

File: drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java (hunk 1 above)
Comment (dupliaka) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: I meant something like  : assertTrue(expressionEvaluator.evaluateUnaryExpression("? = 1*?", BigDecimal.valueOf(2), BigDecimal.class));

Comment (danielezonca) [reply drools-scenario-simulation/drools-scenario-simulation-backend/src/test/java/org/drools/scenariosimulation/backend/expression/DMNFeelExpressionEvaluatorTest.java:52]: Done, added two new tests to cover these additional cases :)

File: drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/utils/ConstantsHolder.java (hunk 2 above)
Comment (danielezonca) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/utils/ConstantsHolder.java]: Good suggestion, done!
